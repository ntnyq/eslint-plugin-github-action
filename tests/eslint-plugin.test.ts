import { ESLint } from 'eslint'
import { describe, expect, it } from 'vitest'
import { resolve } from './internal'

const TEST_CWD = resolve('tests/fixtures/integrations/eslint-plugin')

describe('Integration test', () => {
  it('should lint without errors', async () => {
    const eslint = new ESLint({ cwd: TEST_CWD })
    const files = [
      '.github/workflows/valid.yml',
      '.github/workflows/invalid.yml',
      '.github/workflows/valid.yaml',
      '.github/workflows/invalid.yaml',
    ]
    const results: ESLint.LintResult[] = await eslint.lintFiles(files)

    expect(results.length).toBe(files.length)

    results.forEach(result => {
      expect(result.messages).toMatchSnapshot()
    })
  })
})
