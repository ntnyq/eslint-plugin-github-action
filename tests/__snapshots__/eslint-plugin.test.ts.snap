// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Integration test > should lint without errors 1`] = `[]`;

exports[`Integration test > should lint without errors 2`] = `
[
  {
    "column": 1,
    "endColumn": 34,
    "endLine": 10,
    "line": 1,
    "message": "Require action name to be set.",
    "messageId": "requireActionName",
    "nodeType": "YAMLMapping",
    "ruleId": "github-action/require-action-name",
    "severity": 2,
  },
]
`;

exports[`Integration test > should lint without errors 3`] = `[]`;

exports[`Integration test > should lint without errors 4`] = `
[
  {
    "column": 1,
    "endColumn": 34,
    "endLine": 10,
    "line": 1,
    "message": "Require action name to be set.",
    "messageId": "requireActionName",
    "nodeType": "YAMLMapping",
    "ruleId": "github-action/require-action-name",
    "severity": 2,
  },
]
`;
